### PR TITLE
[MU4] Added opengl buglist

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #8270

### CODE REVIEW

**Comment by wizofaus** [comment]
Unfortunately this makes no difference to the transparent frame or mouse offset issue for me.

**Comment by Eism** [comment]
sad, this hack helped me
I'll think about how to fix the bug

As a temporary solution, you can write in appshel.cpp
QCoreApplication::setAttribute(Qt::AA_UseSoftwareOpenGL);

**Comment by shoogle** [comment]
The PR didn't work for me either, I'm afraid.

> As a temporary solution, you can write in appshel.cpp
QCoreApplication::setAttribute(Qt::AA_UseSoftwareOpenGL);

This works but the app runs slowly when switching tabs or opening dialogs, etc. so not ideal obviously.

**Comment by wizofaus** [comment]
The change to us AA_UseOpenGLES seems to work fine, well done!  No noticeable performance issues either.

**Comment by wizofaus** [comment]
Having said that, it's still different to the borders in other applications, e.g. Visual Studio

image

Whereas MuseScore now looks like

image

So missing that single-pixel darker line.

File: src/appshell/appshell.cpp [hunk 1]
```diff
@@ -70,6 +70,16 @@ int AppShell::run(int argc, char** argv)
         appName  = "MuseScore4";
     }
 
+#ifdef Q_OS_WIN
+    // NOTE: There are some problems with rendering the application window on some integrated graphics processors
+    //       see https://github.com/musescore/MuseScore/issues/8270
+    QCoreApplication::setAttribute(Qt::AA_UseOpenGLES);
+
+    if (!qEnvironmentVariableIsSet("QT_OPENGL_BUGLIST")) {
+        qputenv("QT_OPENGL_BUGLIST", ":/resources/win_opengl_buglist.json");
```
Comment (wizofaus) [inline src/appshell/appshell.cpp:79]: BTW it works fine on my system without this, just setting AA_UseOpenGLES is sufficient.

**Comment by Jojo-Schmitz** [comment]
Isn't this basically a resurrection of https://github.com/musescore/MuseScore/commit/ed4949c72d3f1d8f779561b29186ed59c10a766f ?
Begs the question why/how it got lost for master?

File: src/appshell/appshell.cpp (hunk 1 above)
Comment (Eism) [reply src/appshell/appshell.cpp:79]: yes, but there may be problems with these processors and it is very difficult to understand what the problem is.
insure against them

Comment (Jojo-Schmitz) [reply src/appshell/appshell.cpp:79]: > BTW it works fine on my system without this, just setting AA_UseOpenGLES is sufficient.

See ed4949c as to why it was needed in 3.x, there's no reason to assume it isn't needed in master, as far as I can tell

**Comment by wizofaus** [comment]
It's only the AA_UseOpenGLES that makes a difference for me, the other changes don't seem to be needed (though may well solve other problems).

**Comment by Jojo-Schmitz** [comment]
They sure did in 3.x, and I see no reason to assume them to not be needed for master too?

**Comment by wizofaus** [comment]
3.x used Qt Widgets though and that did a lot of things differently. But honestly I don't know.

**Comment by Jojo-Schmitz** [comment]
It used QML (and therefor QtQuick) too (like for the palettes), just not as much.

**Comment by wizofaus** [comment]
Apparently AA_UseOpenGLES does nothing with Qt 6, something to be aware of if there's any serious consideration of switching.

**Review by shoogle — APPROVED** [review]
Works for me with ` QCoreApplication::setAttribute(Qt::AA_UseOpenGLES);`. We might as well keep the `BUGLIST` stuff if that helped on @Eism's machine.

> it's still different to the borders in other applications, e.g. Visual Studio
> So missing that single-pixel darker line, and missing the app icon from where you can bring up the window menu

The app border on my machine is the same as File Explorer. The window menu appears when I right-click on the title bar so I don't think the app icon is a big loss.

image

I have dual monitors. The app menu is incorrectly scaled on the external non-HDPI monitor but this was already the case before this PR. It is always scaled correctly on the internal HDPI monitor, and also on the external monitor if I disable the internal one.

**Review by RomanPudashkin — APPROVED** [review]

**Comment by vshalkevich** [comment]
Seems fine even with minor bugs

**Pull request merged**